htlcswitch/link.go

```diff
@@ -2035,6 +2049,20 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			"assigning index: %v", msg.PaymentHash[:], index)
 
 	case *lnwire.UpdateFulfillHTLC:
+		if !l.quiescer.canRecvUpdates() {
+			l.fail(
```

Comment:
Hmm, I think we'd still want them to be able to send settle+fail message, I might be misunderstanding the current spec draft though, will double check there.

Reply:
From a game theoretic POV, maybe. Implementing the spec as-written, though, requires that I treat this scenario as a protocol violation.